### PR TITLE
Fix/live 44100 stereo overflow

### LINKED ISSUE #352: [Bug]: openasr live 在44.1 kHz stereo f32下越界panic
### OpenASR version or commit SHA

a946f77a77caf660a6936af3d651242d21297e77

### OS

Ubuntu 26.04

### Rust version

rustc 1.95.0 (59807616e 2026-04-14)

### GPU model and graphics driver version

Intel(R) Iris(R) Xe Graphics (vulkan backend)

### Command run or API request

```shell
RUST_BACKTRACE=1 ./openasr live
```

### Expected behavior

正常进行录音并转录

### Actual behavior

直接panic (见log)

---

询问AI，给出的修复建议是

```diff
diff --git a/crates/openasr-core/src/realtime/capture/core.rs b/crates/openasr-core/src/realtime/capture/core.rs
index d2ad7a4..20b6b28 100644
--- a/crates/openasr-core/src/realtime/capture/core.rs
+++ b/crates/openasr-core/src/realtime/capture/core.rs
@@ -208,7 +208,7 @@ impl CaptureEngine {
             output.push(f32_to_i16(a + (b - a) * fraction));
             self.resample_pos += self.resample_step;
         }
-        let consumed = self.resample_pos.floor() as usize;
+        let consumed = (self.resample_pos.floor() as usize).min(self.resample_input.len());
         if consumed > 0 {
             self.resample_input.drain(0..consumed);
             self.resample_pos -= consumed as f64;
```

我在自己的fork上(https://github.com/CCLL-0x01/openasr/tree/fix/live-44100-stereo-overflow) 尝试修复，通过了github ci (https://github.com/CCLL-0x01/openasr/actions/runs/33228377151), 编译后在本地问题解决。

（我没在其他平台与硬件真机测试，所以没直接pr）

### Logs or output

```text
$ RUST_BACKTRACE=1 ./openasr --version 
openasr 0.1.36 
$ RUST_BACKTRACE=1 ./openasr live 
openasr: warning: local catalog '/home/user/Applications/openasr-0.1.36-linux-x86_64-vulkan/model-registry/catalog.json' epoch 2026081503 is below the epoch floor 2026081504 recorded on this machine; loading it anyway as a degraded boot candidate rather than refusing to start (see docs/CATALOG_COMPATIBILITY.md) 
OpenASR live microphone started on Default Audio Device (f32, 44100 Hz, 2 channel(s)). Streaming partial/final updates are enabled. 
thread 'main' (1770022) panicked at /rustc/59807616e1fa2540724bfbac14d7976d7e4a3860/library/core/src/slice/index.rs:1020:51: range end index 513 out of range for slice of length 512 
stack backtrace: 
0: __rustc::rust_begin_unwind 
1: core::panicking::panic_fmt 
2: core::slice::index::slice_index_fail 
3: openasr_core::realtime::capture::CaptureEngine::push_mono_f32 
4: openasr_core::realtime::capture::CaptureEngine::push_f32_interleaved 
5: openasr::live::LiveCaptureRun::run 
6: openasr::run::{{closure}} 
7: openasr::main::{{closure}} 
8: tokio::runtime::park::CachedParkThread::block_on 
9: openasr::main 
note: Some details are omitted, run with RUST_BACKTRACE=full for a verbose backtrace.
```

### Was OPENASR_HOME customized?

No

### Backend used

native (default, real transcription)

### Native .oasr pack details, if used

_No response_

### Audio sharing confirmation

- [x] I have not attached private audio unless it is safe to share publicly.

### PR DESCRIPTION
**Pull request by CCLL-0x01**
## Summary

This pull request addresses an edge case in the audio resampling logic and adds a new test to ensure correct behavior when processing interleaved stereo audio data in chunks.

## Why

fix #352 

## Changes

- In `CaptureEngine`, updated the calculation of the `consumed` variable to ensure it does not exceed the length of `resample_input`, preventing potential out-of-bounds access during audio resampling (`core.rs`).

- Added a new test, `resampler_handles_44_1khz_stereo_across_512_sample_chunks`, to verify that the resampler correctly handles 44.1kHz stereo audio data provided in 512-sample chunks, matching the callback shape that previously exposed overflows (`tests.rs`).

## Tests run

- [x] `cargo fmt --check`
- [x] `cargo clippy --all-targets -- -D warnings`
- [x] `cargo nextest run --workspace`
- [x] `cargo test --workspace --doc`
- [x] `cargo deny check`

(github actions ci passed)

## Manual smoke checks

```
./openasr live
```

## Model-family changes (when applicable) (not related)

- [ ] I followed the root `AGENTS.md` model-family onboarding entry point and
      ran `cargo xtask family conformance --profile-id <profile-id>`.
- [ ] I stated `core-only`, `staged release candidate`, or `public-ready`; any
      staged/public-ready work completes Model Onboarding Step 5 without
      hand-editing generated catalog/registry truth.
- [ ] Real-weight quality/performance claims have artifact-backed evidence;
      otherwise they are explicitly listed as unverified.

## Docs updated (not related)

- [ ] README or docs updated, if behavior or limitations changed. 
- [ ] No docs overclaim current capabilities. 

## Scope / non-goals

`crate::realtime::capture`

## Artifact confirmation

- [x] I did not commit model weights, large binaries, generated artifacts, secrets, or private audio.
- [x] I did not add vendored runtime sources outside `crates/openasr-core/third_party/openasr-ggml`.
- [x] I did not add unverified model download URLs or fake SHA256 hashes.

## Requirements

- [x] I have read and agree with the [contributing guidelines](../CONTRIBUTING.md) and the AI usage policy in [AGENTS.md](../AGENTS.md).
- [x] I understand every line of this change and can explain it without AI assistance, and I own its maintenance.
- AI usage disclosure: YES, AI for diagnosis and human for code writing and reviewing.

### CODE REVIEW

**Comment by QuintinShaw** [comment]
Thanks for the repro, the 512-frame callback shape, and the consume clamp. That bound is the correct timeline (leftover `pos` continues in the next chunk), not a skip.

I landed it on the current mainline as https://github.com/QuintinShaw/openasr/pull/360 with a named helper and chunked-vs-oneshot identity for 44.1 kHz stereo/mono and 48 kHz / 512. Credit is on the commit.

Closing this PR as superseded by #360. Appreciate the report and the patch.
